src/main/java/ch/njol/skript/sections/EffSecShoot.java

```diff
@@ -0,0 +1,215 @@
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.sections;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.*;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.skript.util.Direction;
+import ch.njol.skript.util.Getter;
+import ch.njol.skript.variables.Variables;
+import ch.njol.util.Kleenean;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Fireball;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Projectile;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.bukkit.projectiles.ProjectileSource;
+import org.bukkit.util.Consumer;
+import org.bukkit.util.Vector;
+import org.eclipse.jdt.annotation.Nullable;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+@Name("Shoot")
+@Description({"Shoots a projectile (or any other entity) from a given entity.",
+	"This can be used as an effect and as a section.",
+	"If it is used as a section, the section is run before the entity is added to the world.",
+	"You can modify the entity in this section, using for example 'event-entity' or 'event-arrow'. ",
+	"Do note that other event values, such as 'player', won't work in this section."})
+@Examples({"shoot an arrow",
+	"make the player shoot a creeper at speed 10",
+	"shoot a pig from the creeper",
+	"",
+	"#Shoot section example",
+	"command /firearrow:",
+	"	trigger:",
+	"		shoot an arrow:",
+	"			ignite event-projectile"})
+@Since("1.4, INSERT VERSION (with section)")
+public class EffSecShoot extends EffectSection {
+
+	public static class ShootEvent extends Event {
+		private final Entity entity;
+
+		public ShootEvent(Entity entity) {
+			this.entity = entity;
+		}
+
+		public Entity getEntity() {
+			return entity;
+		}
+
+		@Override
+		public @NotNull
+		HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+	}
+
+	static {
+		Skript.registerSection(EffSecShoot.class,
+			"shoot %entitydatas% [from %livingentities/locations%] [(at|with) (speed|velocity) %-number%] [%-direction%]",
+			"(make|let) %livingentities/locations% shoot %entitydatas% [(at|with) (speed|velocity) %-number%] [%-direction%]");
+		EventValues.registerEventValue(EffSecShoot.ShootEvent.class, Entity.class, new Getter<Entity, EffSecShoot.ShootEvent>() {
```

Comment:
You should directly reference `ShootEvent.class` rather than `EffSecShoot.ShootEvent.class`